src/python/dxpy/utils/resolver.py

```diff
@@ -1157,18 +1199,69 @@ def resolve_app(path):
     else:
         return desc
 
+def resolve_global_workflow(path):
+    '''
+    :param path: A string which is supposed to identify a global workflow
+    :type path: string
+    :returns: The describe hash of the global workflow object
+    :raises: :exc:`ResolutionError` if it cannot be found
+
+    *path* is expected to have one of the following forms:
+
+    - hash ID, e.g. "globalworkflow-F85Z6bQ0xku1PKY6FjGQ011J"
+    - named ID, e.g. "globalworkflow-myworkflow"
+    - named ID with alias (version or tag), e.g. "globalworkflow-myworkflow/1.2.0"
+    '''
+    desc = get_global_workflow_from_path(path)
+    if desc is None:
+        raise ResolutionError('The given path "' + path + '" could not be resolved to an accessible global workflow')
+    else:
+        return desc
+
+def resolve_global_executable(path):
+    """
+    :param path: A string which is supposed to identify a global executable (app or workflow)
+    :type path: string
+    :returns: The describe hash of the global executable object (app or workflow)
+    :raises: :exc:`ResolutionError` if it cannot be found
+
+    *path* is expected to have one of the following forms:
+
+    - hash ID, e.g. "globalworkflow-F85Z6bQ0xku1PKY6FjGQ011J", "app-FBZ3f200yfzkKYyp9JkFVQ97"
+    - named ID, e.g. "app-myapp", "globalworkflow-myworkflow"
+    - named ID with alias (version or tag), e.g. "app-myapp/1.2.0", "globalworkflow-myworkflow/1.2.0"
+    """
+    # First, check if the prefix is provided, then we don't have to resolve the name
+    if path.startswith('app-'):
+        return resolve_app(path)
+    elif path.startswith('globalworkflow-'):
+        return resolve_global_workflow(path)
+
+    # If the path doesn't include a prefix, we must try describing
+    # as an app and, if that fails, as a global workflow
+    desc = get_app_from_path(path)
+    if not desc:
+        desc = get_global_workflow_from_path(path)
+    if desc is None:
+        raise ResolutionError('The given path "' + path + '" could not be resolved to an accessible global executable (app or workflow)')
+    return desc
+
 def get_exec_handler(path, alias=None):
     handler = None
     def get_handler_from_desc(desc):
         if desc['class'] == 'applet':
             return dxpy.DXApplet(desc['id'], project=desc['project'])
         elif desc['class'] == 'app':
             return dxpy.DXApp(dxid=desc['id'])
-        else:
+        elif desc['class'] == 'workflow':
             return dxpy.DXWorkflow(desc['id'], project=desc['project'])
+        else:
```

Comment:
Minor, may want to explicitly characterize a globalworkflow here in an elif and throw an exception otherwise.

Reply:
Good point, it'll be clearer, thanks!